### PR TITLE
Fixed regression in setting target ports

### PR DESCRIPTION
**Pull request by davidfowl**
- Setting the target port explicitly skips using the DCP expression which stops DCP from allocating endpoints.
- Updated tests

PS: Found while updating samples.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/3796)

### CODE REVIEW

File: src/Aspire.Hosting/ApplicationModel/EndpointReference.cs [hunk 1]
```diff
@@ -182,12 +182,6 @@ public class EndpointReferenceExpression(EndpointReference endpointReference, En
 
     private string? ComputeTargetPort()
     {
-        // We have a target port, so we can return it directly.
-        if (Endpoint.TargetPort is int port)
-        {
-            return port.ToString(CultureInfo.InvariantCulture);
-        }
-
         // There is no way to resolve the value of the target port until runtime. Even then, replicas make this very complex because
```
Comment (ReubenBond) [inline src/Aspire.Hosting/ApplicationModel/EndpointReference.cs:185]: This comment no longer fits

**Review by ReubenBond — APPROVED** [review]
Aside from the now context-lacking comment, LGTM

**Review by ReubenBond — APPROVED** [review]

**Review by mitchdenny — APPROVED** [review]

**Review by karolz-ms — APPROVED** [review]

**Comment by davidfowl** [comment]
I can't reproduce this in release/8.0 nor main... Still investigating.

**Comment by davidfowl** [comment]
OK repro'd on release/8.0!

**Pull request closed without merging**